model/unit_tests.py

```diff
@@ -0,0 +1,181 @@
+""" Unit tests for functions used in the transformer model.
+"""
+
+from absl.testing import parameterized
+import numpy as np
+import tensorflow as tf
+import jax.numpy as jnp
+
+from models import create_relative_ids
+
+
+class UnitTest(tf.test.TestCase, parameterized.TestCase):
+
+  def test_make_relative_ids_enc_only(self):
+    enc_relative_ids =  create_relative_ids(in_length=6, relative_radius=3)
+
+    expected = jnp.array([
+        [3, 2, 1, 0, 0, 0],  #
+        [4, 3, 2, 1, 0, 0],  #
+        [5, 4, 3, 2, 1, 0],  #
+        [6, 5, 4, 3, 2, 1],  #
+        [6, 6, 5, 4, 3, 2],  #
+        [6, 6, 6, 5, 4, 3],  #
+    ])
+    self.assertAllEqual(expected, enc_relative_ids)
+    
+  def test_make_relative_ids_enc_and_dec(self):
+    enc_relative_ids, dec_relative_ids, dec2enc_relative_ids = create_relative_ids(
+                                                    in_length=6, relative_radius=3, 
+                                                    tar_length1=6)
```

Comment:
nit: If you have the bandwidth, I'd suggest using a different value for in_length than for tar_length1 to make sure the dimensions are used in the right axes. For example, set tar_length1 = 5